pkgs/tools/misc/debootstrap/default.nix

```diff
@@ -13,63 +24,43 @@ stdenv.mkDerivation rec {
     sha256 = "117xgrv6mpszyndmsvkn4ynh57b2s40qc68bpmfmxggw58j42klw";
   };
 
-  buildInputs = [ dpkg gettext gawk perl wget ];
+  nativeBuildInputs = [ makeWrapper ];
 
   dontBuild = true;
 
-  # If you have to update the patch for functions a vim regex like this
-  # can help you identify which lines are used to write scripts on TARGET and
-  # which should /bin/ paths should be replaced:
-  # \<echo\>\|\/bin\/\|^\s*\<cat\>\|EOF\|END
   installPhase = ''
-    sed -i \
-      -e 's@/usr/bin/id@id@' \
-      -e 's@/usr/bin/dpkg@${dpkg}/bin/dpkg@' \
-      -e 's@/usr/bin/sha@${coreutils}/bin/sha@' \
-      -e 's@/bin/sha@${coreutils}/bin/sha@' \
-      debootstrap
-
-    for file in functions debootstrap; do
-      substituteInPlace "$file" \
-        --subst-var-by gunzip "$(type -p gunzip)" \
-        --subst-var-by bunzip "$(type -p bunzip)" \
-        --subst-var-by gettext "$(type -p gettext)" \
-        --subst-var-by dpkg "$(type -p dpkg)" \
-        --subst-var-by udpkg "$(type -p udpkg)" \
-        --subst-var-by id "$(type -p id)" \
-        --subst-var-by perl "$(type -p perl)" \
-        --subst-var-by uname "$(type -p uname)" \
-        --subst-var-by wget "$(type -p wget)"
-    done
-
+    runHook preInstall
 
-    sed -i  \
-      -e 's@\<wget\>@${wget}/bin/wget@' \
-      functions
+    substituteInPlace debootstrap \
+      --replace 'CHROOT_CMD="chroot '  'CHROOT_CMD="${coreutils}/bin/chroot ' \
+      --replace 'CHROOT_CMD="unshare ' 'CHROOT_CMD="${utillinux}/bin/unshare ' \
+      --replace /usr/bin/dpkg ${dpkg}/bin/dpkg \
+      --replace '#!/bin/sh' '#!/bin/bash' \
```

Comment:
The interpreter gets patched, yes (if that's what you meant). The issue is that the script relies on some bash-isms which is why change `sh` to `bash` here. I left a comment on the upstream commit here that perhaps makes more sense - https://salsa.debian.org/installer-team/debootstrap/commit/15b20e644b407602288402f97fc348b00921cc54#note_48468 LMK If not!
Sorry if I misunderstood the question and thanks for the merge! Would be nice to have some sort of a backport too. The version on the stable branch is a bit behind but not working so perhaps it would be fine to backport the package updates + this commit?

Reply:
This is interesting because debian also has dash as /bin/sh.